symfony/plugins/orangehrmTimePlugin/Api/EmployeeTimesheetAPI.php

```diff
@@ -324,7 +324,7 @@ public function update(): EndpointResult
             $actionKey = array_flip(TimesheetService::TIMESHEET_ACTION_MAP)[$action];
 
             $allowedActions = $this->getTimesheetService()
-                ->getAllowedWorkflowsForTimesheet($this->getEmpNumber(), $timesheet);
+                ->getAllowedWorkflowsForTimesheet($this->getAuthUser()->getEmpNumber(), $timesheet);
```

Comment:
Not clear

Reply:
that is for work flow state machine, to get the allowed action based on the logged in user role